pkg/api/config/config.go

```diff
@@ -22,8 +22,9 @@ var (
 
 type StorageConfig struct {
 	RootDirectory string
-	GC            bool
 	Dedupe        bool
+	RemoteCache   bool
```

Comment:
If "RemoteCache" is **NOT** explicitly specified, then default behavior is as follows.
* local storage driver -> RemoteCache=false
* remote storage driver (s3) -> RemoteCache=true

ValidateConfig()
* local storage driver, but **explicit** RemoteCache=true, this is not valid, so error out
* maintain pairing (s3 storage, dynamodb), tomorrow, different pairing for (gcp storage, gcp db) etc

Reply:
(local storage, local db) -> dedupe
(remote storage, local db) -> dedupe but not stateful cluster
(remote storage, remote db) -> dedupe and stateful cluster